language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/ExMemory.hs

```diff
@@ -90,20 +92,28 @@ instance (Generic a, GExMemoryUsage (Rep a)) => ExMemoryUsage (GenericExMemoryUs
 class ExMemoryUsage a where
     memoryUsage :: a -> ExMemory -- ^ How much memory does 'a' use?
 
-deriving via (GenericExMemoryUsage (Constant ann)) instance ExMemoryUsage ann => ExMemoryUsage (Constant ann)
 deriving via (GenericExMemoryUsage (Name ann)) instance ExMemoryUsage ann => ExMemoryUsage (Name ann)
-deriving via (GenericExMemoryUsage TypeBuiltin) instance ExMemoryUsage TypeBuiltin
-deriving via (GenericExMemoryUsage (Type TyName ann)) instance ExMemoryUsage ann => ExMemoryUsage (Type TyName ann)
+deriving via (GenericExMemoryUsage (Type TyName uni ann)) instance ExMemoryUsage ann => ExMemoryUsage (Type TyName uni ann)
 deriving via (GenericExMemoryUsage (Builtin ann)) instance ExMemoryUsage ann => ExMemoryUsage (Builtin ann)
 deriving via (GenericExMemoryUsage (Kind ann)) instance ExMemoryUsage ann => ExMemoryUsage (Kind ann)
 deriving via (GenericExMemoryUsage BuiltinName) instance ExMemoryUsage BuiltinName
 deriving via (GenericExMemoryUsage DynamicBuiltinName) instance ExMemoryUsage DynamicBuiltinName
-deriving via (GenericExMemoryUsage (Plain Term)) instance ExMemoryUsage (Plain Term)
-deriving via (GenericExMemoryUsage (WithMemory Term)) instance ExMemoryUsage (WithMemory Term)
+-- TODO: Can we merge these two instances together?
+deriving via (GenericExMemoryUsage (Plain Term uni))
+  instance (Closed uni, uni `Everywhere` ExMemoryUsage) => ExMemoryUsage (Plain Term uni)
+deriving via (GenericExMemoryUsage (WithMemory Term uni))
+  instance (Closed uni, uni `Everywhere` ExMemoryUsage) => ExMemoryUsage (WithMemory Term uni)
 deriving newtype instance ExMemoryUsage ann => ExMemoryUsage (TyName ann)
 deriving newtype instance ExMemoryUsage ExMemory
 deriving newtype instance ExMemoryUsage Unique
 
+instance ExMemoryUsage (Some (In uni)) where
+  memoryUsage _ = 1 -- TODO things like @list (list (list integer))@ take up a non-constant amount of space.
```

Comment:
Maybe link a github issue here.

Reply:
I've created [an issue](https://github.com/input-output-hk/plutus/issues/1861), but I think normally we don't link GitHub issues in the code? Given that we're planning to move to JIRA I certainly hope we won't need to link JIRA issues in the code.

@michaelpj should I link the issue?

Reply:
I do tend to put links to issues in the code, actually. So go for it!

Reply:
Done.